File: ios/Rentivo/Data/LiveDemoRepository.swift
import Foundation

// The live app has no demo switches to flip, so this satisfies `DemoRepository` for
// `AppDependencies.live` while every demo setting stays at its inert default.
@MainActor
public final class LiveDemoRepository: DemoRepository {
  public private(set) var demoSettings = DemoSettings.standard
  public init() {}
  public func failNextOperation() {}
  public func setEmptyMode(_ enabled: Bool) { demoSettings.emptyMode = enabled }
  public func setViewerMode(_ enabled: Bool) { demoSettings.viewerMode = enabled }
  public func setDelayEnabled(_ enabled: Bool) { demoSettings.delayEnabled = enabled }
  public func reset() { demoSettings = .standard }
}

File: android/app/src/main/java/app/rentivo/data/LiveDemoRepository.kt
package app.rentivo.data

/**
 * The live app has no demo switches to flip, so this satisfies [DemoRepository] for the live
 * dependency graph while every demo setting stays at its inert default.
 */
class LiveDemoRepository : DemoRepository {

  override var demoSettings: DemoSettings = DemoSettings.standard
    private set

  override fun failNextOperation() = Unit

  override fun setEmptyMode(enabled: Boolean) {
    demoSettings = demoSettings.copy(emptyMode = enabled)
  }

  override fun setViewerMode(enabled: Boolean) {
    demoSettings = demoSettings.copy(viewerMode = enabled)
  }

  override fun setDelayEnabled(enabled: Boolean) {
    demoSettings = demoSettings.copy(delayEnabled = enabled)
  }

  override fun reset() {
    demoSettings = DemoSettings.standard
  }
}
